### PR TITLE
Add deprecation message for nncf.torch.create_compressed_model

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

Add deprecation massage for `nncf.torch.create_compressed_model`

### Related tickets

152996

### CODE REVIEW

File: nncf/torch/model_creation.py [hunk 1]
```diff
@@ -100,6 +101,11 @@ def create_compressed_model(
         is an instance of CompositeCompressionController) and the model ready for compression parameter training wrapped
         as an object of NNCFNetwork.
     """
+
+    warning_deprecated(
+        "The 'nncf.torch.create_compressed_model' function is deprecated and will be removed in a future release."
```
Comment (MaximProshin) [inline nncf/torch/model_creation.py:106]: The deprecation message includes a recommendation of what to use instead. Let's add it.

**Review by MaximProshin — CHANGES_REQUESTED** [review]

File: nncf/torch/model_creation.py (hunk 1 above)
Comment (AlexanderDokuchaev) [reply nncf/torch/model_creation.py:106]:
```
The 'nncf.torch.create_compressed_model' function is deprecated and will be removed in a future release.
For quantization use 'nncf.quantize', another algorithms are deprecated.
```
ok?

Comment (MaximProshin) [reply nncf/torch/model_creation.py:106]: @AlexKoff88 , we deprecate API here but we have to state if there is an alternative recommended way. For the quantization, it's nncf.quantize() but there is no one for other algorithms. Are you ok if we state them as deprecated?

Comment (AlexKoff88) [reply nncf/torch/model_creation.py:106]: It is a good idea and we can say something like "For QAT, use the new nncf.quantize() API from the example ...". I would omit to mention other algorithms as there is no clarity of what we will keep eventually.

Comment (MaximProshin) [reply nncf/torch/model_creation.py:106]: @AlexanderDokuchaev , please update the warning and then we will merge the PR.

Comment (AlexanderDokuchaev) [reply nncf/torch/model_creation.py]: Updated

**Review by MaximProshin — APPROVED** [review]

**Review by AlexKoff88 — APPROVED** [review]

**Pull request merged**